File: sdks/ruby/lib/novabilling/wallets/types/create_wallet_dto.rb
# frozen_string_literal: true

module Novabilling
  module Wallets
    module Types
      class CreateWalletDto < Internal::Types::Model
        field :customer_id, -> { String }, optional: false, nullable: false, api_name: "customerId"
        field :name, -> { String }, optional: true, nullable: false
        field :currency, -> { String }, optional: false, nullable: false
        field :rate_amount, -> { Integer }, optional: true, nullable: false, api_name: "rateAmount"
        field :paid_credits, -> { Integer }, optional: true, nullable: false, api_name: "paidCredits"
        field :granted_credits, -> { Integer }, optional: true, nullable: false, api_name: "grantedCredits"
        field :expiration_at, -> { String }, optional: true, nullable: false, api_name: "expirationAt"
        field :metadata, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false
        field :created_at, -> { String }, optional: true, nullable: false, api_name: "createdAt"
      end
    end
  end
end

File: sdks/rust/src/api/types/create_wallet_dto.rs
pub use crate::prelude::*;

/// Request type for API operation
#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
pub struct CreateWalletDto {
    #[serde(rename = "customerId")]
    pub customer_id: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub name: Option<String>,
    pub currency: String,
    /// 1 credit = rateAmount in currency
    #[serde(rename = "rateAmount")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub rate_amount: Option<f64>,
    /// Paid credits (purchase)
    #[serde(rename = "paidCredits")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub paid_credits: Option<f64>,
    /// Free credits (grant)
    #[serde(rename = "grantedCredits")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub granted_credits: Option<f64>,
    /// Expiration date (ISO 8601)
    #[serde(rename = "expirationAt")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub expiration_at: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub metadata: Option<HashMap<String, serde_json::Value>>,
    /// Backdate createdAt (ISO 8601). For data imports.
    #[serde(rename = "createdAt")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub created_at: Option<String>,
}
